From a887e9b2cafebae43aabbb7daf10331bde59773b Mon Sep 17 00:00:00 2001
From: "Suresh, Prashanth" <prashanth.suresh@intel.com>
Date: Fri, 3 Mar 2023 14:01:37 +0530
Subject: [PATCH] update media_codecs xml as per MediaSDK_C2

update media_codecs_intel_c2_video.xml as per below PR
https://github.com/projectceladon/MediaSDK_C2/pull/70

Tracked-On:
Signed-off-by: Kothapeta, BikshapathiX <bikshapathix.kothapeta@intel.com>
---
 .../true/media_codecs_intel_c2_video.xml      | 58 ++++++++++---------
 1 file changed, 32 insertions(+), 26 deletions(-)

diff --git a/groups/codec2/true/media_codecs_intel_c2_video.xml b/groups/codec2/true/media_codecs_intel_c2_video.xml
index af080dd..497d244 100644
--- a/groups/codec2/true/media_codecs_intel_c2_video.xml
+++ b/groups/codec2/true/media_codecs_intel_c2_video.xml
@@ -18,64 +18,70 @@
 from AOSP frameworks/av/media/libstagefright/data/media_codecs_google_c2_video.xml
 and updated to vendor media codecs.
 -->
+
+<!--
+    Max block-count = maxHeight / block-size(height) * maxWidth / block-size(W)
+    Max blocks-per-second = Max block-count * frames-per-second
+-->
+
 <Included>
     <Decoders>
 {{#hw_vd_h264}}
-        <MediaCodec name="c2.intel.avc.decoder" type="video/avc">
+	<MediaCodec name="c2.intel.avc.decoder" type="video/avc">
             <!-- profiles and levels:  ProfileHigh : Level52 -->
             <Limit name="size" min="64x64" max="4096x4096" />
             <Limit name="alignment" value="2x2" />
             <Limit name="block-size" value="16x16" />
-            <Limit name="block-count" range="1-32768" /> <!-- max 4096x2048 equivalent -->
+            <Limit name="block-count" range="1-65536" /> <!-- max 4096x4096 equivalent -->
             <Limit name="blocks-per-second" range="1-1966080" />
-            <Limit name="bitrate" range="1-48000000" />
+            <Limit name="bitrate" range="1-40000000" />
             <Limit name="performance-point-3840x2160" value="30" />
             <Feature name="adaptive-playback" />
-        </MediaCodec>
+	</MediaCodec>
 {{/hw_vd_h264}}
 
 {{#hw_vd_h265}}
-        <MediaCodec name="c2.intel.hevc.decoder" type="video/hevc">
+	<MediaCodec name="c2.intel.hevc.decoder" type="video/hevc">
             <!-- profiles and levels:  ProfileMain : MainTierLevel51 -->
             <Limit name="size" min="64x64" max="8192x8192" />
             <Limit name="alignment" value="2x2" />
             <Limit name="block-size" value="8x8" />
-            <Limit name="block-count" range="1-196608" /> <!-- max 4096x3072 -->
-            <Limit name="blocks-per-second" range="1-2000000" />
-            <Limit name="bitrate" range="1-10000000" />
+            <Limit name="block-count" range="1-1048576" /> <!-- max 8192x8192 -->
+            <Limit name="blocks-per-second" range="1-31457280" />
+            <Limit name="bitrate" range="1-40000000" />
             <Limit name="performance-point-3840x2160" value="30" />
             <Feature name="adaptive-playback" />
-        </MediaCodec>
+	</MediaCodec>
 {{/hw_vd_h265}}
 
 {{#hw_vd_vp9}}
-        <MediaCodec name="c2.intel.vp9.decoder" type="video/x-vnd.on2.vp9">
+	<MediaCodec name="c2.intel.vp9.decoder" type="video/x-vnd.on2.vp9">
             <Limit name="size" min="64x64" max="8192x8192" />
             <Limit name="alignment" value="2x2" />
             <Limit name="block-size" value="16x16" />
-            <Limit name="block-count" range="1-16384" />
-            <Limit name="blocks-per-second" range="1-500000" />
+            <Limit name="block-count" range="1-262144" />
+            <Limit name="blocks-per-second" range="1-7864320" />
             <Limit name="bitrate" range="1-40000000" />
             <Limit name="performance-point-3840x2160" value="30" />
-            <Feature name="adaptive-playback" />
-        </MediaCodec>
+	    <Feature name="adaptive-playback" />
+	</MediaCodec>
 {{/hw_vd_vp9}}
 
 {{#hw_vd_vp8}}
-        <MediaCodec name="c2.intel.vp8.decoder" type="video/x-vnd.on2.vp8">
+	<MediaCodec name="c2.intel.vp8.decoder" type="video/x-vnd.on2.vp8">
             <Limit name="size" min="64x64" max="4096x4096" />
             <Limit name="alignment" value="2x2" />
             <Limit name="block-size" value="16x16" />
-            <Limit name="block-count" range="1-16384" />
+            <Limit name="block-count" range="1-131072" />
             <Limit name="blocks-per-second" range="1-500000" />
             <Limit name="bitrate" range="1-40000000" />
             <Limit name="performance-point-3840x2160" value="30" />
             <Feature name="adaptive-playback" />
-        </MediaCodec>
+	</MediaCodec>
 {{/hw_vd_vp8}}
 
 {{#hw_vd_mp2}}
-        <MediaCodec name="c2.intel.mp2.decoder" type="video/mpeg2">
+	<MediaCodec name="c2.intel.mp2.decoder" type="video/mpeg2">
             <Limit name="size" min="64x64" max="2048x2048" />
             <Limit name="alignment" value="2x2" />
             <Limit name="block-size" value="16x16" />
@@ -84,11 +90,11 @@ and updated to vendor media codecs.
             <Limit name="bitrate" range="1-40000000" />
             <Limit name="performance-point-1920x1080" value="30" />
             <Feature name="adaptive-playback" />
-        </MediaCodec>
+	</MediaCodec>
 {{/hw_vd_mp2}}
 
 {{#hw_vd_av1}}
-        <MediaCodec name="c2.intel.av1.decoder" type="video/av01">
+	<MediaCodec name="c2.intel.av1.decoder" type="video/av01">
             <Limit name="size" min="64x64" max="4096x4096" />
             <Limit name="alignment" value="2x2" />
             <Limit name="block-size" value="16x16" />
@@ -97,13 +103,13 @@ and updated to vendor media codecs.
             <Limit name="bitrate" range="1-40000000" />
             <Limit name="performance-point-3840x2160" value="30" />
             <Feature name="adaptive-playback" />
-        </MediaCodec>
+	</MediaCodec>
 {{/hw_vd_av1}}
     </Decoders>
 
     <Encoders>
 {{#hw_ve_h264}}
-        <MediaCodec name="c2.intel.avc.encoder" type="video/avc">
+	<MediaCodec name="c2.intel.avc.encoder" type="video/avc">
             <!-- profiles and levels:  ProfileBaseline : Level41 -->
             <Limit name="size" min="176x144" max="4096x4096" />
             <Limit name="alignment" value="2x2" />
@@ -118,18 +124,18 @@ and updated to vendor media codecs.
 {{/hw_ve_h264}}
 
 {{#hw_ve_h265}}
-        <MediaCodec name="c2.intel.hevc.encoder" type="video/hevc" >
+	<MediaCodec name="c2.intel.hevc.encoder" type="video/hevc" >
             <Limit name="size" min="176x144" max="8192x8192" />
             <Limit name="alignment" value="2x2" />
             <Limit name="block-size" value="16x16" />
             <Limit name="blocks-per-second" range="1-972000" />
             <Limit name="bitrate" range="1-40000000" />
             <Limit name="performance-point-3840x2160" value="30" />
-        </MediaCodec>
+	</MediaCodec>
 {{/hw_ve_h265}}
 
 {{#hw_ve_vp9}}
-	    <MediaCodec name="c2.intel.vp9.encoder" type="video/x-vnd.on2.vp9" >
+	<MediaCodec name="c2.intel.vp9.encoder" type="video/x-vnd.on2.vp9" >
             <Limit name="size" min="128x96" max="8192x8192" />
             <Limit name="alignment" value="2x2" />
             <Limit name="block-size" value="16x16" />
@@ -137,7 +143,7 @@ and updated to vendor media codecs.
             <Limit name="blocks-per-second" range="1-972000" />
             <Limit name="bitrate" range="1-40000000" />
             <Limit name="performance-point-3840x2160" value="30" />
-        </MediaCodec>
+	</MediaCodec>
 {{/hw_ve_vp9}}
     </Encoders>
 </Included>
-- 
2.39.2

